library/src/animationManager.cxx

```diff
@@ -274,3 +274,37 @@ bool animationManager::LoadAtTime(double timeValue)
   return true;
 }
 }
+void f3d::detail::animationManager::CycleAnimation()
+{
+
+  assert(this->Importer);
+  int nIndex = this->Importer->GetNumberOfAnimations();
+  if (nIndex <= 0)
+  {
+    return;
+  }
+  if (this->HasAnimation)
+  {
+    this->AnimationIndex = (this->AnimationIndex + 1) % nIndex;
+  }
+  else
+  {
+    this->AnimationIndex = (this->AnimationIndex + 2) % (nIndex + 1) - 1;
```

Comment:
not sure to follow what this does, if there is no animation, there should not be anything to do ?

Reply:
This one is from the beginning, during an experimental stage, back when I was new to the code base. You previously recommended studying cycle scalars implementation in order to understand how to cycle animations. At the beginning, I was copy-pasting "frankensteining" a rough draft, so I can get a better visual in my head of how everything interconnects and would look in the end. This part happened to be a portion I borrowed from an existing cycle scalars implementation.

You are right. It doesn't need to be there. I'm currently in the process of going back in to clean everything up. Honestly, I was pretty overwhelmed learning about all this new graphics content at the very beginning. Over time, with the assistance of yourself as well as other F3D contributors, I've been learning so much about graphics software. Thanks!

Reply:
No worries! Thanks for your patience :)